### PR TITLE
Add snapshot workflow

### PR DESCRIPTION
**Pull request by sky1045**
SSIA

### CODE REVIEW

**Review by area363 — APPROVED** [review]

File: charts/mothership/templates/snapshot-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    - metadata:
+        name: snapshot-data-volume
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        storageClassName: {{ $.Values.storageClassName }}
+        resources:
+          requests:
+            storage: 100Gi   
+
+    # This spec contains two templates: hello-hello-hello and whalesay
```
Comment (eseiker) [inline charts/mothership/templates/snapshot-workflow.yaml:34]:
```suggestion
```

File: charts/mothership/templates/snapshot-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+      - - name: fetch-genesis-
+          template: fetch-genesis 
+      - - name: fetch-snapshot-
+          template: fetch-snapshot 
+      - - name: genesis-init-
+          template: genesis-init
+      - - name: op-stack-
+          template: op-stack
+      - - name: create-snapshot-
+          template: create-snapshot
```
Comment (eseiker) [inline charts/mothership/templates/snapshot-workflow.yaml:49]: are `- -` and hypens after step names intended?

File: charts/mothership/templates/snapshot-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+      resources:
+        requests:
+          memory: 2Gi
+      {{- with $.Values.node.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+
+    # This is the same template as from the previous example
```
Comment (eseiker) [inline charts/mothership/templates/snapshot-workflow.yaml:58]: i don't think so,,,
```suggestion
```

**Review by eseiker — APPROVED** [review]
seems okay, with some minor comments

File: charts/mothership/templates/snapshot-workflow.yaml (hunk 2 above)
Comment (sky1045) [reply charts/mothership/templates/snapshot-workflow.yaml]: https://argo-workflows.readthedocs.io/en/stable/workflow-concepts/#steps
i think we can use both `- -` and `-`. the former seems to make it execute multiple steps parallel.
and we dont need the last hyphen as it isnt used for prefix now

**Review by eseiker — APPROVED** [review]

**Review by area363 — APPROVED** [review]

**Pull request merged**